assets/js/script.js

```diff
@@ -1,97 +1,101 @@
 $(document).ready(function() {
-    var letter1Choice = "black"
-    var letter2Choice = "black"
+  var el = {
+    input:          $('#letters-input'),
+    letterTabs:     $('.letter-color-letter'),
+    letterTab1:     $('#letter1'),
+    letterTab2:     $('#letter2'),
+    lettersWrapper: $('.letters-wrapper'),
+    swatch:         $('.swatch-image-border'),
+  }
 
-    $(window).resize(function() {
-        var newLetterHeight = $(".letters-wrapper ").height()
-        $("#bagLetter1").css({ display: "block", height: newLetterHeight + "px" })
-        $("#bagLetter2").css({ display: "block", height: newLetterHeight + "px" })
+  Object.keys(el).forEach(function(key) {
+    if( !el[key].length ) { console.warn(`No element present for ${key}`); }
+  })
 
+  var letter1Choice = "black"
+  var letter2Choice = "black"
 
-    })
+  $(window).resize(function() {
+    el.letters.height(
+      el.lettersWrapper.height()
+    )
+  })
+
+  el.letterTabs.on('click', function() {
+    el.letterTabs.removeClass("active")
+    $(this).addClass("active")
+  })
 
-    $(".letter-color-letter").click(function() {
-        $(".letter-color-letter.active").removeClass("active")
+  el.letterTab1.on('click', function() {
+    el.swatch.removeClass('active')
+    el.swatch.each(function() {
+      if ($(this).data("material-swatch") == letter1Choice) {
         $(this).addClass("active")
+      }
     })
+  })
 
-    $("#letter1").click(function() {
-        $(".swatch-image-border.active").removeClass("active")
-        $(".swatch-image-border").each(function() {
-            if ($(this).data("material-swatch") == letter1Choice) {
-                $(this).addClass("active")
-            }
-        })
-    })
-    $("#letter2").click(function() {
-        $(".swatch-image-border.active").removeClass("active")
-        $(".swatch-image-border").each(function() {
-            if ($(this).data("material-swatch") == letter2Choice) {
-                $(this).addClass("active")
-            }
-        })
+  el.letterTab2.on('click', function() {
+    el.swatch.removeClass('active')
+    el.swatch.each(function() {
+      if ($(this).data("material-swatch") == letter2Choice) {
+        $(this).addClass("active")
+      }
     })
-    $("#letters-input").on("input", function() {
+  })
 
+  el.input.on("input", function() {
+    var $inputText = $(this).val()
 
-        var $inputText = $(this).val()
-        var newHeight = $(".letters-wrapper ").height()
+    if( $inputText.length > 0 ) {
+      $("#bagLetter2").hide()
+      renderLetter(1, $inputText.charAt(0), letter1Choice)
```

Comment:
This is the meat of the refactor, extracting the code for rendering a single letter and putting it in its own function. That way if/when we change how we're rendering letters, we don't have to update it in 4 different places. It also makes this event handler a little more readable as to what the logic is doing.